app/src/main/java/com/chouten/app/ui/views/home/HomePageViewModel.kt

```diff
@@ -35,94 +37,89 @@ class HomePageViewModel(
+        if(shouldReset){
+            this.errors = 0;
         }
 
+        _homeResults.clear()
+        viewModelScope.launch { loadHomePage() }
     }
 
-    private suspend fun runHomeFns(homeFns: List<ModuleModel.ModuleCode.ModuleCodeblock>) {
-        homeFns.forEach { homeFn ->
-            println("HOME FN: $homeFn")
-            viewModelScope.launch {
-                if (!webview.load(
-                        homeFn,
-                    )
-                ) {
-                    isLoading = false
-                    return@launch
-                }
-
-                val res = webview.inject(homeFn)
-                if (res.isBlank()) {
-                    PrimaryDataLayer.enqueueSnackbar(
-                        SnackbarVisualsWithError(
-                            "Something went wrong while loading the home page.",
-                            false
-                        )
-                    )
-                    isLoading = false
-                    return@launch
-                }
-
-                try {
-                    val results =
-                        Mapper.parse<ModuleResponse<List<HomeResult>>>(
-                            res
-                        )
-                    println("RESULTS: $results")
-                    _homeResults.clear()
-                    _homeResults.addAll(results.result)
-                    isLoading = false
-                    webview.updateNextUrl(results.nextUrl)
-                } catch (e: Exception) {
-                    isLoading = false
-                    PrimaryDataLayer.enqueueSnackbar(
-                        SnackbarVisualsWithError(
-                            e.localizedMessage ?: "Error parsing home page results.", isError = true
-                        )
-                    )
-                    e.printStackTrace()
-                }
+    fun getCode(): String{
+        val currentModule = ModuleLayer.selectedModule;
+        val subtype = currentModule?.subtypes?.getOrNull(0);
+        var code: String;
 
-            }
+        if(subtype == null){
+            throw Exception("Subtype not found");
+        }else{
+            val tempCode = currentModule.code?.get(subtype)?.home?.getOrNull(0)?.code;
 
-        }
+            if(tempCode == null){
+                throw Exception("code not found");
+            }else{
+                code = tempCode;
+            }
+        }   
+        return code;
     }
 
+    private suspend fun loadHomePage() {
+        Log.d("here1234", "loadHomePage called");
+        Log.d("here1234", this.errors.toString());
+
+        _loadedModule.value = ModuleLayer.selectedModule
+        val homeModule = ModuleLayer.selectedModule ?: return
+        
+        isLoading = true;
+        val code = this.getCode();
+
+        val webviewPayload = mapOf<String, String>(
+            "action" to "homepage"
+        );
+
+        webview.load(code, JSONObject(webviewPayload).toString());
```

Comment:
JSONObject may be substituted with `Mapper.json.encodeToString`

Reply:
TODO added.